addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js

```diff
@@ -118,18 +114,23 @@ const ColorPaletteWidget = Widget.extend({
         this.colorToColorNames = {};
         this.el.querySelectorAll('button[data-color]').forEach(elem => {
             const colorName = elem.dataset.color;
-            const color = ColorpickerDialog.normalizeCSSColor(this.style.getPropertyValue('--' + colorName).trim());
             const $color = $(elem);
             $color.addClass('bg-' + colorName);
-            if (this.selectedColor && (this.selectedColor === colorName || this.selectedColor === color)) {
-                $color.addClass('selected');
-            }
             this.colorNames.push(colorName);
             if (!elem.classList.contains('d-none')) {
+                const color = ColorpickerDialog.normalizeCSSColor(this.style.getPropertyValue('--' + colorName).trim());
                 this.colorToColorNames[color] = colorName;
             }
         });
 
+        // Select selected Color
+        if (this.options.selectedColor) {
+            this.selectedColor = ColorpickerDialog.normalizeCSSColor(this.options.selectedColor);
+            this.selectedColor = this.colorToColorNames[this.selectedColor] || this.selectedColor;
+            const selectedButton = this.el.querySelector(`button[data-color="${this.selectedColor}"], button[style*="background-color:${this.selectedColor};"]`);
+            selectedButton && selectedButton.classList.add('selected');
```

Comment:
Update your linter

Reply:
Updated with this : https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines
I see no error/warning.
You mean I can not use `&&` to verify if a variable is set and do an action ?

Reply:
Then your linter does not work at all, otherwise `"no-unused-expressions": ["warn"],` would warn you here.

Reply:
new npm package babel-eslint was not installed...